crates/libs/core/src/com_object.rs

```diff
@@ -0,0 +1,274 @@
+use crate::{AsImpl, IUnknown, IUnknownImpl, Interface, InterfaceRef};
+use core::ptr::NonNull;
+use std::borrow::Borrow;
+
+/// Identifies types that can be placed in `ComObject`.
+///
+/// This trait links types that can be placed in `ComObject` with the types generated by the
+/// `#[implement]` macro. The `#[implement]` macro generates implementations of this trait.
+/// The generated types contain the vtable layouts and refcount-related fields for the COM
+/// object implementation.
+///
+/// This trait is an implementation detail of the Windows crates.
+/// User code should not deal directly with this trait.
+pub trait ComImpl {
```

Comment:
`#[doc(hidden)]` for this and `ComObjectInterface`?

Reply:
In some cases user code needs to import traits, and I think concealing too much of the machinery makes it hard to use the Windows crate.  There are a handful of methods that are meaningful to user code, like `IUnknownImpl::from_inner_ref`, `get_impl`, etc.  I think it's best not to conceal these from docs.